Framework/PythonInterface/mantid/simpleapi.py

```diff
@@ -302,6 +302,39 @@ def fitting_algorithm(f):
     When applied to a function definition this decorator replaces its code with code of
     function 'wrapper' defined below.
     """
+    def gather_fitting_returns(function_name, lhs, algm):
```

Comment:
I have a few objections:
1. An error made in the python layer should be fixed in the python layer and not spread it to others.
2. Fit specific changes were made in the Fit (fitting) specific area of simpleapi.
3. `OutputNIterators` is a permanent property (not optional as `OutputNormalisedCovarianceMatrix` is) and should be declared by `init()`.